enumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    each do |item|
+      yield item
+    end
+
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    each do |item|
+      yield item, index
+      index += 1
+    end
+
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    results_array = []
+    each do |item|
+      next unless yield item
+
+      results_array << item
+    end
+    results_array
+  end
+
+  def my_all?(pattern = nil)
+    result = true
+    if block_given?
+      each do |item|
+        result = false unless yield item
+      end
+    elsif pattern
+      each do |item|
+        result = false unless pattern.match(item)
+      end
+    else
+      each do |item|
+        result = false unless item
+      end
+    end
+    result
+  end
+
+  def my_none?(pattern = nil)
+    result = true
+    if block_given?
+      each do |item|
+        result = false if yield item
+      end
+    elsif pattern
+      each do |item|
+        result = false if pattern.match(item)
+      end
+    else
+      each do |item|
+        result = false if item
+      end
+    end
+    result
+  end
```

Comment:
### Class
- [ ] Try it passing a class
```ruby
   
  [1, 2i, 3.14 r].my_none?(Numeric)
# Answer should be true because `r ` not belongs to Numeric class, should belong to all elements. Check original none method
  [hello world 1]%w.my_none?(String)
# Answer should be true because `1` not belongs to String class, should belong to all elements. Check original none method
```

## Parameter also should accept numbers not only regexp
![image](https://user-images.githubusercontent.com/24902525/90536019-21896e00-e141-11ea-84ac-c417dd39c86d.png)

-  Line number 66
```ruby
        result = false if pattern.match(item)
```
- Do this test
`[hellow world].my_none?(3)` should be equal to `[hellow world].none?(3)`